src/Stratis.Bitcoin/P2P/SelfEndpointTracker.cs

```diff
@@ -11,6 +20,35 @@ public class SelfEndpointTracker : ISelfEndpointTracker
         /// <summary>Hashset to hold the endpoints currently known to be itself.</summary>
         private readonly ConcurrentHashSet<IPEndPoint> knownSelfEndpoints = new ConcurrentHashSet<IPEndPoint>();
 
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        /// <summary>External IP address of the node.</summary>
+        public IPEndPoint MyExternalAddress { get; set; }
+
+        /// <summary>Whether IP address of the node is final or can be updated.</summary>
+        public bool IsMyExternalAddressFinal { get; set; } = false;
+
+        /// <summary>Peer score of external IP address of the node.</summary>
+        public int MyExternalAddressPeerScore { get; set; }
+
+        /// <summary>
+        /// Initializes an instance of the self endpoint tracker.
+        /// </summary>
+        public SelfEndpointTracker() : this(new LoggerFactory())
```

Comment:
You don't want this constructor. It would be only used by tests and for those they can call it with the correct log factory anyway